common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -293,6 +302,14 @@ def _is_in_bulk_operation(self, course_key, ignore_case=False):
         """
         return self._get_bulk_ops_record(course_key, ignore_case).active
 
+    def send_pre_publish_signal(self, bulk_ops_record, course_id):
+        """
+        Send a signal just before items are published in the course.
+        """
+        signal_handler = getattr(self, "signal_handler", None)
+        if signal_handler and bulk_ops_record.has_publish_item:
+            signal_handler.send("pre_publish", course_key=course_id)
```

Comment:
@jimabramson: I was just poking through the signal handling code, and I noticed that as-written, we won't send a pre-publish signal if we're not in a bulk-op. That seems problematic to me.